File: 8.string-to-integer-atoi/main.py
# coding:utf-8


class Solution:
    def myAtoi(self, str: str) -> int:
        new_str = str.strip()
        flag = False
        if len(new_str) == 0:
            return 0
        if new_str[0] == '-':
            flag = True
            new_str = new_str[1:]
        elif new_str[0] == '+':
            new_str = new_str[1:]
        if len(new_str) == 0:
            return 0
        num = 0
        s = set(["1", "2", "3", "4", "5", "6", "7", "8", "9", "0"])
        for i in new_str:
            if i in s:
                num = num * 10 + int(i)
                if num > 2147483647:
                    if flag:
                        num = 2147483648
                    else:
                        num = 2147483647
                    break
            else:
                break
        if flag:
            num = -num
        return num


if __name__ == '__main__':
    print(Solution().myAtoi("   2147483649"))

File: 8.string-to-integer-atoi/main.js
/**
 * @param {string} str
 * @return {number}
 */
var myAtoi = function (str) {
    new_str = str.trim()
    num = 0
    flag = false
    const set1 = new Set(["1", "2", "3", "4", "5", "6", "7", "8", "9", "0"]);
    for (let i = 0; i < new_str.length; i++) {
        if ((i===0)&&(new_str[0] === '-')){
            flag = true
        } else if ((i===0)&&(new_str[0] === '+')) {
            flag = false
        } else if (set1.has(new_str[i])) {
            num = num * 10 + parseInt(new_str[i])
        } else {
            break
        }
        if (num > 2147483647) {
            if (flag) {
                num = 2147483648
            } else {
                num = 2147483647
            }
            break
        }
    }
    if (flag) {
        num = -num
    }
    return num
};

(function (main) {
    console.log(myAtoi("2147483649"))
}())